sycl/source/detail/jit_compiler.cpp

```diff
@@ -591,11 +601,12 @@ updatePromotedArgs(const ::jit_compiler::SYCLKernelInfo &FusedKernelInfo,
       // argument is later on passed to the kernel.
       const size_t SizeAccField =
           sizeof(size_t) * (Req->MDims == 0 ? 1 : Req->MDims);
-      // Compute the local size and use it for the range parameters.
-      auto LocalSize = getLocalSize(NDRange, Req,
-                                    (PromotedToPrivate) ? Promotion::Private
-                                                        : Promotion::Local);
-      range<3> AccessRange{1, 1, LocalSize.value()};
+      // Compute the local size and use it for the range parameters (only
+      // relevant for local promotion).
```

Comment:
If this is only relevant for local promotion, should we still get here when private promoting?

Reply:
My understanding is that we still need to rewrite the arguments when private promoting, it's just that the concrete size is irrelevant in that case.

Reply:
We do use the "LocalSize" when private promoting. If we have a buffer of range `1024` and a kernel of range `512`, the local size will be 2, meaning we will allocate 2 elements per work-item in private memory.

Reply:
Yes, but IIUC, when this code here is run, the entire fusion pipeline including internalization has already completed, and private-internalized accessor args have been marked unused, so I don't think the runtime evaluates the size set here.

My reason for making this distinction is that we don't have anything meaningfully resembling a "user range" for the *fused* kernel at this point that I could pass to the `getLocalSize` helper. So it's either local-size-for-local-promotion (which is unaffected by RR), or dont-care-local-size-for-private-promotion.

Reply:
True, my bad. LGTM as is :+1: